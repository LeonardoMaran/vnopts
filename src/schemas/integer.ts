import { Utils, ValidateResult } from '../types';
import { isInt } from '../utils';
import { NumberSchema } from './number';

export class IntegerSchema extends NumberSchema {
  public expected() {
    return 'an integer';
  }
  public validate(value: unknown, utils: Utils): ValidateResult {
    return (
      utils.normalizeValidateResult(super.validate(value, utils), value) ===
        true && isInt(value as number)
    );
  }
}
