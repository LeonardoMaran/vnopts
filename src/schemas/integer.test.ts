import { eachHandler } from '../../tests/__helpers__/utils';
import { IntegerSchema } from './integer';

const name = '<key>';
const value = 1;

describe.each`
  parameters | input                | output               | hasWarnings
  ${{}}      | ${{ [name]: value }} | ${{ [name]: value }} | ${false}
  ${{}}      | ${{ [name]: 1.5 }}   | ${Error}             | ${false}
`('', eachHandler<IntegerSchema>(IntegerSchema, { name }));
