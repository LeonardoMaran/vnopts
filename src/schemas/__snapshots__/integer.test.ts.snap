// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`schema {"name":"<key>"} with input {"<key>":1.5} 1`] = `"Invalid \\"<key>\\" value. Expected an integer, but received 1.5."`;
